float32 left_angle
float32 left_radius
float32 right_angle
float32 right_radius
